app/derivative-services/jp2_derivative_service.rb

```diff
@@ -114,7 +120,28 @@ def use
   end
 
   def filename
-    return Pathname.new(file_object.io.path) if file_object.io.respond_to?(:path) && File.exist?(file_object.io.path)
+    Pathname.new(file_object.disk_path)
```

Comment:
I wonder why these error checks were in here. your change is definitely an improvement, but it makes me wonder if there were errors? Not sure how there could be an io without a path.

Reply:
This code is from the Valkyrie MVP and existed before `disk_path` was a required property in the Valkyrie API.